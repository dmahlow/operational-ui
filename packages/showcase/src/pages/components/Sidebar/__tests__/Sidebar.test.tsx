import * as React from "react"
import { BrowserRouter as Router } from "react-router-dom"
import { render } from "enzyme"

import SidebarPage, { fetch } from "../Sidebar"

describe("Sidebar Page", () => {
  it("Should render correctly", () => {
    expect(
      render(
        <Router>
          <SidebarPage />
        </Router>
      )
    ).toMatchSnapshot()
  })

  it("Should have a proper timeout for the async test", async() => {
    const fetchedVal = await fetch("hi")
    expect(fetchedVal).toBe("hi")
  })
})
