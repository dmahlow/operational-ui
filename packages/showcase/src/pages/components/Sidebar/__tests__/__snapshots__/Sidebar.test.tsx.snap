// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Page Should render correctly 1`] = `
<div>
  <div
    class="css-by30gj"
  >
    Sidebar
  </div>
  <p>
    The sidebar is a dynamic list-style navigational element to be used in cases with a large number of list-style actionable items. This component involves composition of two constituent elements. Namely,
  </p>
  <ul>
    <li>
      <a
        href="#sidebar-item"
      >
        SidebarItem
      </a>
    </li>
    <li>
      <a
        href="#sidebar-link"
      >
        SidebarLink
      </a>
    </li>
  </ul>
  <div
    style="margin-bottom:32px;"
  />
  <h4>
    Usage
  </h4>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          t]
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
  <div
    style="margin-bottom:32px;"
  />
  <div
    class="css-by30gj"
    id="sidebar-item"
  >
    SidebarItem
  </div>
  <h4>
    An expandable group of SidebarLinks, with added asynchronous functionality.
  </h4>
  <table
    class="css-81lp1j"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          label
        </td>
        <td>
          The label of the SidebarItem.
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
      <tr>
        <td>
          open
        </td>
        <td>
          Is the item open or closed by default?
        </td>
        <td>
          false
        </td>
        <td>
          boolean
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
      <tr>
        <td>
          onClick
        </td>
        <td>
          A function to pass to the item that executes before the item expands. If a function returning a Promise is passed in, the item only expands after the Promise resolves.
        </td>
        <td>
          <pre>
            () =&gt; this.open = !this.open
          </pre>
        </td>
        <td>
          func
        </td>
        <td>
          âœ…
        </td>
      </tr>
    </tbody>
  </table>
  <p
    style="margin-top:16px;margin-bottom:32px;"
  >
    <strong>
      Note: This component is wrapped with 
      <a
        href="/tooltips"
      >
        withTooltip
      </a>
       and thus exposes all of the props that such components do.
    </strong>
  </p>
  <div
    class="css-by30gj"
    id="sidebar-link"
  >
    SidebarLink
  </div>
  <p>
    A link, but with onClick instead of href.
  </p>
  <table
    class="css-81lp1j"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          to
        </td>
        <td>
          Created to work with react-router, this wraps the children in a &lt;Link&gt; to your route.
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          onClick
        </td>
        <td>
          A function called on click of this component, to be used instead of the \`to\` prop to do more than just navigate.
        </td>
        <td />
        <td>
          func
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          symbol
        </td>
        <td>
          A symbol to display at the right-hand side of the link, such as a \`%\` sign to suggest a unit of measure.
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          color
        </td>
        <td>
          Different links can have different colors to communicate different use-cases or purposes. This can be a hex value, or a named color in your theme.
        </td>
        <td>
          The primary color of your theme.
        </td>
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
    </tbody>
  </table>
  <p
    style="margin-top:16px;margin-bottom:32px;"
  >
    <strong>
      Note: This component is wrapped with 
      <a
        href="/tooltips"
      >
        withTooltip
      </a>
       and thus exposes all of the props that such components do.
    </strong>
  </p>
</div>
`;
